MonoGame.Framework/Graphics/Texture2D.DirectX.cs

```diff
@@ -180,7 +181,9 @@ private void PlatformConstruct(int width, int height, bool mipmap, SurfaceFormat
                         for (var row = 0; row < rows; row++)
                         {
                             int i;
-                            for (i = row * rowSize / elementSizeInByte; i < (row + 1) * rowSize / elementSizeInByte; i++)
+                            // pre-evaluate non-dependant condition (32 bits of memory vs (1 add, 1 multiply and 1 divide per iteration))
+                            int condition =  (row + 1) * rowSize / elementSizeInByte;
+                            for (i = row * rowSize / elementSizeInByte; i < condition; i++)
                                 data[i + startIndex] = stream.Read<T>();
```

Comment:
the `+ startIndex` could be moved into the initial value of i and condition, saving yet another add per loop iteration